desc/objectives/_stability.py

```diff
@@ -445,176 +451,118 @@ def __init__(
             jac_chunk_size=jac_chunk_size,
         )
 
-        errorif(
-            np.asarray(self._rho).size > 1,
-            ValueError,
-            "BallooningStability objective only works on a single surface. "
-            "To optimize multiple surfaces, use multiple instances of the objective.",
-        )
-
-    def build(self, eq=None, use_jit=True, verbose=1):
+    def build(self, use_jit=True, verbose=1):
         """Build constant arrays.
 
         Parameters
         ----------
-        eq : Equilibrium, optional
-            Equilibrium that will be optimized to satisfy the Objective.
         use_jit : bool, optional
             Whether to just-in-time compile the objective and derivatives.
         verbose : int, optional
             Level of output.
 
         """
-        eq = self.things[0]
+        self._iota_keys = ["iota", "iota_r", "shear", "a"]
 
-        # we need a uniform grid to get correct surface averages for iota
+        eq = self.things[0]
         iota_grid = LinearGrid(
-            rho=self._rho,
+            # to compute length scale quantities correctly
+            rho=np.append(self._rho, 1) if self._add_lcfs else self._rho,
```

Comment:
is it really that much more expensive to create a dedicated grid on the LCFS to compute length scales?

Reply:
I don't understand the question because that is what we are doing here

Reply:
this is adding a new surface to an existing grid which then requires some weird indexing, rather than just using a dedicated grid for the length qtys

Reply:
Oh. This is what I thought to do first. I did not benchmark

Reply:
but it wouldn't actually be any extra flops I think. Whether its 1 grid with extra surface or 2 grids, the total number of nodes is the same.

Reply:
Kind attention